presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -266,26 +266,33 @@ private static PlanNodeStatsEstimate filterStatsForUnknownExpression(PlanNodeSta
                     .collect(ImmutableList.toImmutableList());
 
             if (!valuesEqualityStats.stream().allMatch(Optional::isPresent)) {
-                return visitExpression(node, context);
+                return Optional.empty();
             }
 
             PlanNodeStatsEstimate statsSum = valuesEqualityStats.stream()
                     .map(Optional::get)
                     .reduce(filterForFalseExpression().get(), PlanNodeStatsEstimateMath::addStatsAndSumDistinctValues);
 
             if (isNaN(statsSum.getOutputRowCount())) {
-                return visitExpression(node, context);
+                return Optional.empty();
             }
 
-            Symbol inValueSymbol = Symbol.from(node.getValue());
-            SymbolStatsEstimate symbolStats = input.getSymbolStatistics(inValueSymbol);
-            double notNullValuesBeforeIn = input.getOutputRowCount() * (1 - symbolStats.getNullsFraction());
+            Optional<Symbol> inValueSymbol = asSymbol(node.getValue());
```

Comment:
This looks like an obfuscation. Why don't just do

```
if(node.getValue() instanceof SymbolReference){
 ....
} else {
 ....
}
```

Reply:
I have different opinion on this. Maybe because `instanceof`s scare me :)

Reply:
You can always introduce an utility method, `isSymbolReference`. But i think we just do `instanceof` in other parts of our code.

My point basically that there are simple cases. When your expression is a simple symbol reference, and when the expression is a complex expression. But it took me a while to figure this out.

Reply:
You might be right. This sounds a bit as a matter of taste. And I'm reluctant to do any dispensable changes in this particular class at this point, as I don't want to accidentally break something resolving conflicts during next rebase.